pkg/core/resources/apis/meshmultizoneservice/status_updater.go

```diff
@@ -0,0 +1,139 @@
+package meshmultizoneservice
+
+import (
+	"context"
+	"reflect"
+	"sort"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+
+	meshmzservice_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshmultizoneservice/api/v1alpha1"
+	meshservice_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshservice/api/v1alpha1"
+	"github.com/kumahq/kuma/pkg/core/resources/manager"
+	"github.com/kumahq/kuma/pkg/core/runtime/component"
+	"github.com/kumahq/kuma/pkg/core/user"
+	core_metrics "github.com/kumahq/kuma/pkg/metrics"
+	util_maps "github.com/kumahq/kuma/pkg/util/maps"
+)
+
+type StatusUpdater struct {
+	roResManager manager.ReadOnlyResourceManager
+	resManager   manager.ResourceManager
+	logger       logr.Logger
+	metric       prometheus.Summary
+	interval     time.Duration
+}
+
+var _ component.Component = &StatusUpdater{}
+
+func NewStatusUpdater(
+	logger logr.Logger,
+	roResManager manager.ReadOnlyResourceManager,
+	resManager manager.ResourceManager,
+	interval time.Duration,
+	metrics core_metrics.Metrics,
+) (component.Component, error) {
+	metric := prometheus.NewSummary(prometheus.SummaryOpts{
+		Name:       "component_mzms_status_updater",
+		Help:       "Summary of MeshMultizoneService Updater component",
+		Objectives: core_metrics.DefaultObjectives,
+	})
+	if err := metrics.Register(metric); err != nil {
+		return nil, err
+	}
+	return &StatusUpdater{
+		roResManager: roResManager,
+		resManager:   resManager,
+		logger:       logger,
+		metric:       metric,
+		interval:     interval,
+	}, nil
+}
+
+func (s *StatusUpdater) Start(stop <-chan struct{}) error {
+	s.logger.Info("starting")
+	ticker := time.NewTicker(s.interval)
+	ctx := user.Ctx(context.Background(), user.ControlPlane)
+
+	for {
+		select {
+		case <-ticker.C:
+			start := time.Now()
+			if err := s.updateStatus(ctx); err != nil {
+				s.logger.Error(err, "could not update status of mesh multizone services")
+			}
+			s.metric.Observe(float64(time.Since(start).Milliseconds()))
```

Comment:
don't we want to have metric with fails? It is easier to monitor and add alerting on metric than on logs

Reply:
Hm, we don't do it usually. We could but at this point maybe it make sense to do this as "generic measurable ticker component". Adding this metric and writing ticker component is already a boilerplate. Let me create an issue. I'd like not to add it to this PR

Reply:
https://github.com/kumahq/kuma/issues/10671